sklearn/calibration.py

```diff
@@ -827,7 +827,11 @@ def loss_grad(AB):
             sample_weight=sample_weight,
         )
         loss = l.sum()
-        grad = np.array([-g @ F, -g.sum()])
+        # TODO: Remove casting to np.float64 when minimum supported SciPy is 1.11.2
+        # With SciPy >= 1.11.2, the LBFGS implementation will cast to float64
+        # https://github.com/scipy/scipy/pull/18825. 
+        # Here we cast to float64 to support SciPy < 1.11.2
```

Comment:
This is gonna be very hard to find when we update minimum scipy requirement to `>=1.11.2`. Usually we have these in `fixes.py`. If we don't want to put it there, we should at least have a comment there about this.

Reply:
Note that if we forget to remove it this is not dramatic since the inner SciPy `astype` will be no-op.